conftest.py

```diff
@@ -103,6 +103,18 @@ def get_dispatch_core_type():
     return dispatch_core_type
 
 
+def get_dispatch_core_config(device_params):
+    import ttnn
+
+    dispatch_core_type = get_dispatch_core_type()
+    dispatch_core_axis = device_params.pop(
+        "dispatch_core_axis",
+        ttnn.DispatchCoreAxis.COL if os.environ["ARCH_NAME"] == "blackhole" else ttnn.DispatchCoreAxis.ROW,
```

Comment:
Damn, is there no other way to the arch at this point?

I know with the ttnn API, you have to `CreateDevice` first to get it which you can't do yet because you need the dispatch config :(

Reply:
I will approve for now to unblock, but consider using the `silicon_arch_name` fixture to get the arch name so we can at least environment variable usage in one place.

Reply:
I think both `dispatch_core_type` and `dispatch_core_config` should just be passed in through pytests, using the `device_params`. I have a comment in `get_dispatch_core_type` that says to remove that function.

Reply:
let's have a cleanup in following pr, since changing dispatch_core_type to pass in from device params is beyond the scope of this pr

Reply:
I see, can this env flag be more explicit? Like `TT_DISPATCH_AXIS=col`? Because with this current implementation you can export `ARCH_NAME=blackhole` on a grayskull machine to get a col dispatch behaviour?

Reply:
in the future pr

Reply:
We definitely should remove dispatch core type env flag, currently waiting for models team to clean up their tests.

Reply:
Please open an issue for cleanup and we can put it on runtime team's board, then I'll approve